File: car-management-ui/src/services/CarService.js
import axios from "axios";

const URL = "http://localhost:8080/api/v1/car";

class CarService {
  addCar(car) {
    return axios.post(URL, car);
  }

  getCar(id) {
    return axios.get(URL + "/" + id);
  }
  getCars() {
    return axios.get(URL);
  }

  deleteCar(id) {
    return axios.delete(URL + "/" + id);
  }
  editCar(id, car) {
    return axios.put(URL + "/" + id, car);
  }
}

export default new CarService();

File: car-management-api/src/main/java/com/shakir/carmanagement/service/CarService.java
package com.shakir.carmanagement.service;

import com.shakir.carmanagement.model.Car;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface CarService {
    Boolean deleteCar(Long id);

    Car addCar(Car car);

    List<Car> getCars();

    Car editCar(Long id,Car car);

    Car getCar(Long id);
}
